components/organisms/EditResolverModalContent.tsx

```diff
@@ -0,0 +1,149 @@
+import { chain } from "@/lib/wallet/wallet-config";
+import { setResolver } from "@ensdomains/ensjs/wallet";
+import { Button, Input } from "@ensdomains/thorin";
+import { useState } from "react";
+import toast from "react-hot-toast";
+import { Address, createWalletClient, custom, isAddress } from "viem";
+
+interface EditResolverModalContentProps {
+  name: string;
+  currentResolverAddress: Address;
+  closeModal: () => void;
+  onRecordsEdited?: () => void;
+}
+
+export const EditResolverModalContent = ({
+  currentResolverAddress,
+  closeModal,
+  name,
+}: EditResolverModalContentProps) => {
+  const [resolverAddress, setResolverAddress] = useState<string>(
+    currentResolverAddress
+  );
+  const [transactionHash, setTransactionHash] = useState("");
+  const [transactionSuccess, setTransactionSuccess] = useState(false);
+  const [isLoading, setIsloading] = useState(false);
+
+  const handleSaveAction = async () => {
+    // Create a wallet client for sending transactions to the blockchain
+    const walletClient = createWalletClient({
+      chain: chain,
+      transport: custom(window.ethereum),
+    });
+
+    const [address] = await walletClient.getAddresses();
+
+    if (isAddress(resolverAddress)) {
+      setIsloading(true);
+      try {
+        const hash =
+          address &&
+          (await setResolver(walletClient, {
+            name: name,
+            contract: "nameWrapper",
+            resolverAddress: resolverAddress,
+            account: address,
+          }));
+        setTransactionHash(hash);
+        setTransactionSuccess(true);
+      } catch {
+        () => {
+          toast.error("an error occured");
+        };
+      }
+      setIsloading(false);
+    } else {
+      toast.error("Your resolver must be an address");
+    }
+  };
+
+  return (
+    <div className="w-[480px] border rounded-xl overflow-hidden">
+      <div className="py-5 px-6 flex justify-between w-full bg-gray-50 border-b font-semibold text-black">
+        Edit Resolver
+      </div>
+      <div className="bg-white text-black border-b border-gray-200 p-6 flex flex-col gap-4">
+        {transactionSuccess ? (
+          <>
+            <p className="text-[72px]"> 🎉</p>
+            <div>
+              <p className="text-lg">
+                <span className="font-bold">Congratulations!</span>
+              </p>
+              <p>Update resolver transaction sent</p>
+            </div>
+
+            <p className="text-sm">
+              Check your transaction:{" "}
+              <a
+                className="text-gray-500 underline hover:text-gray-700 transition-colors duration-200"
+                target="_blank"
+                href={`https://sepolia.etherscan.io/tx/${transactionHash}`}
```

Comment:
In these cases it will be very good to have notices that updates are needed once Mainnet support is wanted... Otherwise we will have to deep dive into the whole repo to find every place that needs support... A doc comment or the already implemented support are possibilities!